libpromises/mod_custom.c

```diff
@@ -564,13 +564,22 @@ static void PromiseModule_AppendAllAttributes(
             // Evaluated by agent and not sent to module, skip
             continue;
         }
-        char *const value = RvalToString(attribute->rval);
-        PromiseModule_AppendAttribute(module, name, value);
-        free(value);
+
+        if ((attribute->rval.type == RVAL_TYPE_SCALAR) || (attribute->rval.type == RVAL_TYPE_LIST))
+        {
+            JsonElement *value = RvalToJson(attribute->rval);
```

Comment:
Is this freed somewhere?

Reply:
Yes, it is added to the message json, which is freed on send:

```
DESTROY_AND_NULL(JsonDestroy, module->message);
```